py3status/modules/volume_status.py

```diff
@@ -66,45 +69,37 @@
 """
 
 import re
-import shlex
+from os import devnull
 
 from subprocess import check_output, call
 
 
-class Py3status:
-    """
-    """
-    # available configuration parameters
-    button_down = 0
-    button_mute = 0
-    button_up = 0
-    cache_timeout = 10
-    channel = 'Master'
-    device = 'default'
-    format = u'♪: {percentage}%'
-    format_muted = u'♪: muted'
-    threshold_bad = 20
-    threshold_degraded = 50
-    volume_delta = 5
+class AudioBackend():
+    def __init__(self, device, channel):
+        self.device = device
+        self.channel = channel
 
-    # compares current volume to the thresholds, returns a color code
-    def _perc_to_color(self, string):
-        try:
-            value = int(string)
-        except ValueError:
-            return self.py3.COLOR_BAD
+    def run_cmd(self, cmd):
+        with open(devnull, 'wb') as dn:
+            return call(cmd, stdout=dn, stderr=dn)
 
-        if value < self.threshold_bad:
-            return self.py3.COLOR_BAD
-        elif value < self.threshold_degraded:
-            return self.py3.COLOR_DEGRADED
-        else:
-            return self.py3.COLOR_GOOD
+    def get_volume(self):
+        raise NotImplemented
 
-    # return the format string formatted with available variables
-    def _format_output(self, format, percentage):
-        text = self.py3.safe_format(format, {'percentage': percentage})
-        return text
+    def volume_up(self, delta):
+        raise NotImplemented
+
+    def volume_down(self, delta):
+        raise NotImplemented
+
+    def toggle_mute(self):
+        raise NotImplemented
```

Comment:
I think these methods add little as we will get a AttributeError exception anyway.